src/lib/one-graph/cli-netlify-graph.js

```diff
@@ -291,36 +293,91 @@ const runPrettier = async (filePath) => {
+ * @param {GraphQL.GraphQLSchema} input.schema The schema to use when generating the functions and their types
+ * @param {string} input.schemaId The id of the schema to use when fetching Graph data
+ * @param {string} input.operationsDoc The GraphQL operations doc to use when generating the functions
+ * @param {Record<string, NetlifyGraph.ExtractedFunction>} input.functions The parsed queries with metadata to use when generating library functions
+ * @param {Record<string, NetlifyGraph.ExtractedFragment>} input.fragments The parsed queries with metadata to use when generating library functions
+ * @param {CodegenHelpers.GenerateRuntimeFunction} input.generate
+ * @param {(message: string) => void=} input.logger A function that if provided will be used to log messages
  * @returns {Promise<void>} Void, effectfully writes the generated library to the filesystem
  */
-const generateFunctionsFile = async ({ fragments, functions, logger, netlifyGraphConfig, operationsDoc, schema }) => {
-  const { clientSource, typeDefinitionsSource } = await NetlifyGraph.generateFunctionsSource(
+const generateRuntime = async ({
+  fragments,
+  functions,
+  generate,
+  logger,
+  netlifyGraphConfig,
+  operationsDoc,
+  schema,
+  schemaId,
+}) => {
+  const runtime = await NetlifyGraph.generateRuntime({
+    GraphQL,
     netlifyGraphConfig,
     schema,
     operationsDoc,
-    functions,
+    operations: functions,
     fragments,
-  )
+    generate,
+    schemaId,
+  })
 
-  ensureNetlifyGraphPath(netlifyGraphConfig)
-  const implementationResolvedPath = path.resolve(...netlifyGraphConfig.netlifyGraphImplementationFilename)
-  fs.writeFileSync(implementationResolvedPath, clientSource, 'utf8')
-  const implementationRelativePath = path.relative(process.cwd(), implementationResolvedPath)
-  logger && logger(`Wrote ${chalk.cyan(implementationRelativePath)}`)
-
-  const typeDefinitionsResolvedPath = path.resolve(...netlifyGraphConfig.netlifyGraphTypeDefinitionsFilename)
-  fs.writeFileSync(typeDefinitionsResolvedPath, typeDefinitionsSource, 'utf8')
-  const typeDefinitionsRelativePath = path.relative(process.cwd(), typeDefinitionsResolvedPath)
-  logger && logger(`Wrote ${chalk.cyan(typeDefinitionsRelativePath)}`)
-  runPrettier(path.resolve(...netlifyGraphConfig.netlifyGraphImplementationFilename))
-  runPrettier(path.resolve(...netlifyGraphConfig.netlifyGraphTypeDefinitionsFilename))
+  // ensureNetlifyGraphPath(netlifyGraphConfig)
+  runtime.forEach((file) => {
+    const implementationResolvedPath = path.resolve(...file.name)
+    fs.writeFileSync(implementationResolvedPath, file.content, 'utf8')
+    const implementationRelativePath = path.relative(process.cwd(), implementationResolvedPath)
+    logger && logger(`Wrote ${chalk.cyan(implementationRelativePath)}`)
+    runPrettier(path.resolve(...file.name))
+  })
+
+  // const typeDefinitionsResolvedPath = path.resolve(...netlifyGraphConfig.netlifyGraphTypeDefinitionsFilename)
```

Comment:
I assume we don't need to keep this around anymore?

Reply:
Nice catch! Good to delete that code.